### PR TITLE
API-566: updated filter works on product variant if its product models were updated

### PR DESCRIPTION
**Pull request by willy-ahva**
Only tests here, to be sure it works well.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | OK
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/AbstractProductTestCase.php [hunk 1]
```diff
@@ -131,9 +131,9 @@ protected function assertListResponse(Response $response, $expected)
      */
     protected function assertSameProducts(array $expectedProduct, $identifier)
     {
-        $product = $this->getFromTestContainer('pim_catalog.repository.product')->findOneByIdentifier($identifier);
+        $product = $this->get('pim_catalog.repository.product')->findOneByIdentifier($identifier);
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/AbstractProductTestCase.php]: why ?

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/SuccessListVariantProductDeltaIntegration.php [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+                "scope":null
+            }
+        ]
+    }
+}
+JSON;
+        $clientPatch->request('PATCH', 'api/rest/v1/product-models/prod_mod_optB', [], [], [], $updates);
+        $this->assertSame(Response::HTTP_NO_CONTENT, $clientPatch->getResponse()->getStatusCode());
+
+        $this->waitUntilJobsHaveBeenConsume();
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/SuccessListVariantProductDeltaIntegration.php]: havebeenconsumeD

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/SuccessListVariantProductDeltaIntegration.php [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    {
+        $jobLauncher = new JobLauncher(static::$kernel);
+        while ($jobLauncher->hasJobInQueue()) {
+            $jobLauncher->launchConsumerOnce();
+        }
+    }
+
+    private function getDiscriminantDatetime(): \Datetime
+    {
+        sleep(1);
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/SuccessListVariantProductDeltaIntegration.php]: first sleep is usless no?

Comment (ahocquard) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/SuccessListVariantProductDeltaIntegration.php]: Ok Seen IRL: prevent to have the same date as the fixtures.

Comment (willy-ahva) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/SuccessListVariantProductDeltaIntegration.php]: seen IRL

**Pull request merged**